### PR TITLE
Viewer theme can be switched between dark or light theme

### PR DESCRIPTION
**Pull request by ruchimotwaniBC**
Implemented the theme changing dropdown in Settings dialog. 

The default theme of the application is taken from config.json and if not then default theme is according to users system. 
image

Changes in MarkdownPopover are to support the link color for dark theme.

### CODE REVIEW

**Review by forman — COMMENTED** [review]
@ruchimotwaniBC , looks good, thanks!

Could you please change the implementation logic as follows:

1. Change the control state's `currentAppTheme` type to `string`.
2. If `Config.instance.branding.themeName` is set, initialize control state's `currentAppTheme` to its value.
3. Only if `Config.instance.branding.themeName` is NOT set, allow for changing the theme in the Settings.

**Review by clarasb — COMMENTED** [review]
Reviewed from a user-perspective:

The switch between themes works really well. 
But it seems to have no effect on the apprerance of an editor used in the following features:

1. **Import Places**
2. Sidebar -> Metadata -> **JSON Format**

**Comment by ruchimotwaniBC** [comment]
> Reviewed from a user-perspective:
> 
> The switch between themes works really well. But it seems to have no effect on the apprerance of an editor used in the following features:
> 
> 1. **Import Places**
> 2. Sidebar -> Metadata -> **JSON Format**


Ok I check. Thanks @clarasb

**Review by forman — CHANGES_REQUESTED** [review]

**Review by clarasb — APPROVED** [review]
The feature looks great!

(tested from a user-perspective)

File: src/components/SettingsDialog.tsx [hunk 1]
```diff
@@ -42,6 +42,8 @@ import {
   LocateMode,
   TimeSeriesChartType,
   TimeAnimationInterval,
+  APPLICATION_THEMES,
+  ApplicationThemes,
```
Comment (forman) [inline src/components/SettingsDialog.tsx:46]: Why two of them?

File: src/util/branding.ts [hunk 2]
```diff
@@ -79,7 +79,7 @@ export interface Branding {
   windowTitle: string;
   windowIcon: string | null;
   compact: boolean;
-  themeName: "dark" | "light";
+  themeName: "light";
```
[the remark this answers is not in the document]
Comment (forman) [reply src/util/branding.ts:82]: Will fix that.

File: src/states/controlState.ts [hunk 3]
```diff
@@ -160,6 +166,7 @@ export interface ControlState {
   exportPlacesAsCollection: boolean;
   exportZipArchive: boolean;
   exportFileName: string;
+  themeMode: string;
```
Comment (forman) [inline src/states/controlState.ts:169]: `"light" | "dark" | "system"` - I will fix that

**Review by forman — APPROVED** [review]
Thanks @ruchimotwaniBC

**Pull request merged**